tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/Unroller.scala

```diff
@@ -146,25 +197,32 @@ class Unroller( tracker : TransformationTracker ) extends TlaModuleTransformatio
   }
 
   override def apply( module : TlaModule ) : TlaModule = {
-    val operDecls = module.operDeclarations
+    // Before unrolling we transform all recursive operators into parameter-normal form
+    // This reduces the number of inlining calls when operator arguments aren't primitive
+    val parameterNormalizer = ParameterNormalizer( nameGenerator, tracker, ParameterNormalizer.DecisionFn.recursive )
```

Comment:
This makes me wonder about the API for the `ParameterNormalizer`. Perhaps instead of requiring client code to access these `DecisionFn` functions, it should provide methods like `ParameterNormalizer.recursive`, `ParameterNormalizer.all`?

Reply:
If I understand this correctly, the behavior you describe is that `ParameterNormalizer.recursive(x, y)` is equivalent to `ParameterNormalizer(x,y, DecisionFn.recursive)`? 
If so, I disagree, because this gives us more leeway in the future, w.r.t. introducing custom decision functions. For example, the current implementation allows us (or any other user) to seamlessly call normalization only on nullary operators, or only on operators from one particular module or even only on exactly one named operator. Having modes that we have to pre-program into the normalizer seems much more restrictive.

Reply:
I wasn't suggesting remove the definition of `apply`, just that having to call with this predefined functions internally is clunky, and since you are providing them, it seems like it would be a more elegant API to write method to support that application. No biggie here tho.